File: AuthStore.Lib/Models/AuthModel.cs
namespace KolibSoft.AuthStore.Models;

public class AuthModel
{
    public CredentialModel Credential { get; set; } = null!;
    public IEnumerable<PermissionModel> Permissions { get; set; } = null!;
    public string AccessToken { get; set; } = string.Empty;
    public string RefreshToken { get; set; } = string.Empty;
}

File: docs/lib/models/auth_model.js
import { CatalogueStatics } from "../modules/catalogue.js";
import { CredentialModel } from "./credential_model.js";
import { PermissionModel } from "./permission_model.js";

class AuthModel {

    /** @type {CredentialModel} */
    credential;

    /** @type {PermissionModel[]} */
    permissions;

    /** @type {string} */
    accessToken;

    /** @type {string} */
    refreshToken;

    constructor(json) {
        this.credential = json?.credential ?? new CredentialModel({ id: CatalogueStatics.NoGuid });
        this.permissions = json?.permissions ?? [];
        this.accessToken = json?.accessToken ?? "";
        this.refreshToken = json?.refreshToken ?? "";
    }

}

export {
    AuthModel
}